pkg/controller/account/ec2.go

```diff
@@ -168,24 +182,96 @@ func (r *ReconcileAccount) InitializeRegion(
 	}
 
 	err = r.BuildAndDestroyEC2Instances(reqLogger, account, awsClient, instanceInfo, managedTags, customerTags)
-
 	if err != nil {
 		createErr := fmt.Sprintf("Unable to create instance in region: %s", region)
 		controllerutils.LogAwsError(reqLogger, createErr, nil, err)
 		// Notify Error channel that this region has errored and to move on
 		ec2Errors <- regionInitializationError{ErrorMsg: createErr, Region: region}
-
 		return err
 	}
 
 	successMsg := fmt.Sprintf("EC2 instance created and terminated successfully in region: %s", region)
 
+	// if vpcID exists then make sure we delete it
+	if sampleVPCID != "" {
+		err = deleteVpc(reqLogger, awsClient, sampleVPCID)
+		if err != nil {
+			deleteVpcErr := fmt.Sprintf("Unable to delete VPC with ID: %s", sampleVPCID)
+			controllerutils.LogAwsError(reqLogger, deleteVpcErr, nil, err)
+		}
+	}
+
 	// Notify Notifications channel that an instance has successfully been created and terminated and to move on
 	ec2Notifications <- successMsg
 
 	return nil
 }
 
+// createVpc creates a vpc and returns the VpcId
+func createVpc(reqLogger logr.Logger, client awsclient.Client, account *awsv1alpha1.Account, managedTags []awsclient.AWSTag, customerTags []awsclient.AWSTag) (string, error) {
+	tags := awsclient.AWSTags.BuildTags(account, managedTags, customerTags).GetEC2Tags()
+	input := &ec2.CreateVpcInput{
+		CidrBlock: aws.String(sampleCIDR),
+		TagSpecifications: []*ec2.TagSpecification{
+			{
+				ResourceType: &awsv1alpha1.InstanceResourceType,
+				Tags:         tags,
+			},
+			{
+				ResourceType: &awsv1alpha1.VolumeResourceType,
+				Tags:         tags,
+			},
+		},
+	}
+	result, err := client.CreateVpc(input)
+	if err != nil {
+		vpcErr := fmt.Sprintf("Error while attempting to create VPC: %v", *result.Vpc.VpcId)
+		controllerutils.LogAwsError(reqLogger, vpcErr, nil, err)
+		return "", awsv1alpha1.ErrFailedToCreateVpc
+	}
+	return *result.Vpc.VpcId, nil
+}
+
+// deleteVpc deletes a vpc and returns err
+func deleteVpc(reqLogger logr.Logger, client awsclient.Client, vpcIDtoDelete string) error {
+	_, err := client.DeleteVpc(&ec2.DeleteVpcInput{
+		VpcId: aws.String(vpcIDtoDelete),
+	})
+	if err != nil {
+		vpcErr := fmt.Sprintf("Error while attempting to delete VPC: %s", vpcIDtoDelete)
+		controllerutils.LogAwsError(reqLogger, vpcErr, nil, err)
+		return awsv1alpha1.ErrFailedToDeleteVpc
+	}
```

Comment:
We might want to add some kind of backoff/retry logic.  There's a chance we might not be able to delete the VPC right away while we wait for the EC2 instance to terminate completely, and we don't want to leave the VPC around as the limit for VPCs on new accounts defaults to 5, so that would get used up pretty quickly.